### PR TITLE
Rolling and recreate deployment flows in ODC

### PR DESCRIPTION
**Pull request by Preeticp**
This PR:

- Adds 2 modules for Recreate and Rolling updates in the Deployment Assembly
- Updates includes in the assembly.
- This applies to the 4.3 branch
- This has been reviewed by SME and QE

### CODE REVIEW

File: modules/odc-starting-recreate-deployment.adoc [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+//
+// * applications/deployments/deployment-strategies.adoc
+
+[id="odc-starting-recreate-deployment_{context}"]
+= Starting a Recreate deployment using the Developer perspective
+
+You can switch the deployment strategy from the default Rolling update to a Recreate update using the *Developer* perspective in the web console.
+
+.Prerequisites
+* Ensure that you are in the *Developer* perspective of the web console
```
Comment (ahardin-rh) [inline modules/odc-starting-recreate-deployment.adoc]: missing period at the end of the first bullet point

File: modules/odc-starting-recreate-deployment.adoc [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+* Ensure that you have created an application using the *Add* view and see it deployed in the *Topology* view.
+
+.Procedure
+
+To switch to a recreate update strategy and to upgrade an application:
```
Comment (ahardin-rh) [inline modules/odc-starting-recreate-deployment.adoc]: Recreate (to keep consistent with other usage)

File: modules/odc-starting-recreate-deployment.adoc [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+. In the *Actions* drop-down menu, select *Edit Deployment Config* to see the deployment config details of the application.
```
Comment (ahardin-rh) [inline modules/odc-starting-recreate-deployment.adoc]: s/config/configuration

File: modules/odc-starting-recreate-deployment.adoc [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+. In the YAML editor, change the `spec.strategy.type` to `Recreate` and click *Save*.
+. In the *Topology* view, select the node to see the *Overview* tab in the side panel. The *Update Strategy* is now set to *Recreate*.
+. Use the *Actions* drop-down menu to select *Start Rollout* to start an update using the recreate strategy. The recreate strategy first terminates pods for the older version of the application and then spins up pods for the new version.
```
Comment (ahardin-rh) [inline modules/odc-starting-recreate-deployment.adoc]: Pods (per https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#api-object-formatting)

Comment (ahardin-rh) [reply modules/odc-starting-recreate-deployment.adoc]: Recreate

File: modules/odc-starting-rolling-deployment.adoc [hunk 5]
```diff
@@ -0,0 +1,24 @@
+// Module included in the following assemblies:
+//
+// * applications/deployments/deployment-strategies.adoc
+
+[id="odc-starting-rolling-deployment_{context}"]
+= Starting a Rolling deployment using the Developer perspective
+
+.Prerequisites
+* Ensure that you are in the *Developer* perspective of the web console
```
Comment (ahardin-rh) [inline modules/odc-starting-rolling-deployment.adoc]: Missing period

File: modules/odc-starting-rolling-deployment.adoc [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+. In the *Topology* view of the *Developer* perspective, click on the application node to see the *Overview* tab in the side panel. Note that the *Update Strategy* is set to the default *Rolling* strategy.
+. In the *Actions* drop-down menu, select *Start Rollout* to start a rolling update. The rolling deployment spins up the new version of the application and then terminates the old one.
++
+image::odc-rolling-update.png[Rolling Update]
+
+.Additional resources
+
+* xref:../applications/application-life-cycle-management/odc-creating-applications-using-developer-perspective.adoc#odc-creating-applications-using-developer-perspective[Creating and deploying applications on {product-title} using the *Developer* perspective]
+* xref:../applications/application-life-cycle-management/odc-viewing-application-composition-using-topology-view.adoc#odc-viewing-application-composition-using-topology-view[Viewing the applications in your project, verifying their deployment status, and interacting with them in the *Topology* view].
```
Comment (ahardin-rh) [inline modules/odc-starting-rolling-deployment.adoc]: Possibly unnecessary period at the end of the second bullet point

**Comment by Preeticp** [comment]
@ahardin-rh I have fixed all the comments and squashed the commits, can you merge this, please?

**Pull request merged**